src/poll_manager.ts

```diff
@@ -0,0 +1,149 @@
+import type { StreamChat } from './client';
+import type {
+  CreatePollData,
+  DefaultGenerics,
+  ExtendableGenerics,
+  PollResponse,
+  PollSort,
+  QueryPollsFilters,
+  QueryPollsOptions,
+} from './types';
+import { Poll } from './poll';
+import { FormatMessageResponse } from './types';
+import { formatMessage } from './utils';
+
+export class PollManager<SCG extends ExtendableGenerics = DefaultGenerics> {
+  private client: StreamChat<SCG>;
+  private pollCache = new Map<string, Poll<SCG>>();
+  private unsubscribeFunctions: Set<() => void> = new Set();
+
+  constructor({ client }: { client: StreamChat<SCG> }) {
+    this.client = client;
+  }
+
+  get data(): Map<string, Poll<SCG>> {
+    return this.pollCache;
+  }
+
+  public fromState = (id: string) => {
+    return this.pollCache.get(id);
+  };
+
+  public registerSubscriptions = () => {
+    if (this.unsubscribeFunctions.size) {
+      // Already listening for events and changes
+      return;
+    }
+
+    this.unsubscribeFunctions.add(this.subscribeMessageNew());
+    this.unsubscribeFunctions.add(this.subscribePollUpdated());
+    this.unsubscribeFunctions.add(this.subscribePollClosed());
+    this.unsubscribeFunctions.add(this.subscribeVoteCasted());
+    this.unsubscribeFunctions.add(this.subscribeVoteChanged());
+    this.unsubscribeFunctions.add(this.subscribeVoteRemoved());
+  };
+
+  public unregisterSubscriptions = () => {
+    this.unsubscribeFunctions.forEach((cleanupFunction) => cleanupFunction());
+    this.unsubscribeFunctions.clear();
+  };
+
+  public createPoll = async (poll: CreatePollData<SCG>) => {
+    const { poll: createdPoll } = await this.client.createPoll(poll);
+
+    return new Poll({ client: this.client, poll: createdPoll });
+  };
+
+  public getPoll = async (id: string) => {
+    const { poll } = await this.client.getPoll(id);
+
+    this.setOrOverwriteInCache(poll, true);
+
+    return this.fromState(id);
+  };
+
+  public queryPolls = async (filter: QueryPollsFilters, sort: PollSort = [], options: QueryPollsOptions = {}) => {
```

Comment:
`StreamChat.queryThreads` returns list of `Thread` instances - maybe we could do the same here for general API consistency?

The same goes for state [hydration](https://github.com/GetStream/stream-chat-js/blob/71153c8818347e3a7780d57f1df67f75e6f27a0b/src/thread.ts#L146):

```tsx
poll.hydrateState(differentPollInstanceButSameId)
```

Reply:
Well, `client.queryPolls` should return the raw response in my opinion in case someone needs it (they'll have a way to do it). If they're reaching for reactive state though, they would fetch through `pollManager.queryPolls(...)` - this was the general idea. We can of course move the instantiation there too and simply make sure the cache is populated in `pollManager`, but it seemed like losing an API that way.